src/platform/platform_misc.c

```diff
@@ -68,29 +69,61 @@ static int windows_system(const char *cmd)
 {
     PROCESS_INFORMATION p_info;
     STARTUPINFOW s_info;
-    DWORD ReturnValue;
+    HANDLE hp, p_stderr;
+    DWORD handleInfo, flags, ret, err = ERROR_SUCCESS;
 
     memset(&s_info, 0, sizeof(s_info));
     memset(&p_info, 0, sizeof(p_info));
     s_info.cb = sizeof(s_info);
+    s_info.dwFlags = STARTF_USESTDHANDLES;
 
-    char_os *utf16cmd;
-    utf16cmd = caml_stat_strdup_to_os(cmd);
-    if (CreateProcessW(NULL, utf16cmd, NULL, NULL, FALSE, CREATE_NO_WINDOW, NULL, NULL, &s_info, &p_info))
-    {
+    s_info.hStdInput = INVALID_HANDLE_VALUE;
+
+    /* If needed, duplicate stderr to make sure it is inheritable */
+    p_stderr = GetStdHandle(STD_ERROR_HANDLE);
+    if (p_stderr == INVALID_HANDLE_VALUE) {
+        err = GetLastError(); goto ret1;
+    }
+    if (! GetHandleInformation(p_stderr, &handleInfo)) {
+        err = GetLastError(); goto ret1;
+    }
+    hp = GetCurrentProcess();
+    if (! (handleInfo & HANDLE_FLAG_INHERIT)) {
+        hp = GetCurrentProcess();
+        if (! DuplicateHandle(hp, p_stderr, hp, &(s_info.hStdError),
+                              0, TRUE, DUPLICATE_SAME_ACCESS)) {
+            err = GetLastError(); goto ret1;
+        }
+    } else {
+        s_info.hStdError = p_stderr;
+    }
+
+    /* Redirect stdout to stderr */
+    s_info.hStdOutput = s_info.hStdError;
+
+    flags = CREATE_NO_WINDOW | CREATE_UNICODE_ENVIRONMENT;
+    char_os *utf16cmd = caml_stat_strdup_to_os(cmd);
```

Comment:
```suggestion
    WCHAR *utf16cmd = caml_stat_strdup_to_utf16(cmd);
```
(it is equivalent, but a bit clearer since this is Windows-specific code)